### PR TITLE
Fix auth state recovery during server errors

### PR DESCRIPTION
**Pull request by delano**
## Problem

Users were incorrectly logged out and redirected to signin when encountering server errors (500 responses). This created a confusing experience where a temporary server issue would destroy their session context and force re-authentication.

## Solution

Implemented three-layered authentication state recovery that preserves user sessions during server errors while maintaining security:

1. **httpOnly session cookie** - Server-validated security anchor that can't be manipulated by client-side code
2. **Server `had_valid_session` flag** - Authoritative signal from backend confirming valid session existed before error
3. **Client sessionStorage `ots_auth_state`** - Client-side expectation tracker for recovery coordination

The recovery logic uses an AND gate - all three conditions must be true to preserve auth state, preventing false positives while allowing legitimate recovery.

## Changes

**Backend (Ruby):**
- Added `had_valid_session` to authentication serializer with httpOnly cookie validation
- Fixed missing `sess` in view_vars initialization (critical enabler for recovery logic)
- Added Rhales schema definitions for auth state tracking

**Frontend (TypeScript/Vue):**
- Enhanced `authStore` with recovery logic and detailed logging (+82 lines)
- Improved router guards with comprehensive auth validation (+53 lines)
- Better cookie detection with proper parsing
- New test suite for auth store recovery scenarios (+77 lines)
- SessionStorage coordination for auth state persistence

## Testing

- Added comprehensive authStore tests covering recovery scenarios
- Verified httpOnly cookie validation
- Tested error boundary cases (missing cookies, cleared storage, etc.)

## Files Changed
12 files: 425 insertions, 52 deletions

**Impact:** Production bug fix improving user experience during transient server errors.

### CODE REVIEW

File: apps/web/core/views/serializers/authentication_serializer.rb [hunk 1]
```diff
@@ -25,6 +25,13 @@ def self.serialize(view_vars, _i18n)
 
         output['cust'] = cust.safe_dump
 
+        # Check if there was a valid session at the time of this response
+        # This is crucial for error pages where authenticated=false but the user
+        # had a valid session. The frontend uses this to avoid incorrect logouts.
+        # A valid session has 'external_id' present (customer identifier in session)
+        sess = view_vars['sess']
+        output['had_valid_session'] = !!(sess && !sess.empty? && !sess['external_id'].to_s.empty?)
```
[the remark this answers is not in the document]
Comment (delano) [reply apps/web/core/views/serializers/authentication_serializer.rb:33]: - Assessment: False positive - Rack session middleware already validates sessions before this code runs
- Rationale: The presence of sess['external_id'] confirms Rack validated the session cookie. No redundant validation needed.

File: src/stores/authStore.ts [hunk 2]
```diff
@@ -113,22 +113,66 @@ export const useAuthStore = defineStore('auth', () => {
...
+        'likely server error page, preserving auth state'
+      );
+      isAuthenticated.value = true;
+    } else {
+      // Normal flow: trust window state
+      // Regardless of what the value is, if it isn't exactly true, it's false.
+      // i.e. unlimited ways to fail, only one way to succeed.
+      console.log('✅ Using server auth state (normal flow)');
+      isAuthenticated.value = inputValue === true;
+    }
```
[the remark this answers is not in the document]
Comment (delano) [reply src/stores/authStore.ts:160]: - Assessment: Low priority preference, not critical
- Rationale: Current inline logic is clear and scoped appropriately for this PR
